src/molecule_plugins/podman/playbooks/create.yml

```diff
@@ -144,52 +131,56 @@
-        {% if item.dns_servers is defined %}--dns="{{ item.dns_servers | join(',') }}"{% endif %}
-        {% if item.env is defined %}{% for i, k in item.env.items() %}--env={{ i }}={{ k }} {% endfor %}{% endif %}
-        {% if item.restart_policy is defined %}
-        --restart={{ item.restart_policy }}{% if item.restart_retries is defined %}:{{ item.restart_retries }}{% endif %}{% endif %}
-        {% if item.tty is defined %}--tty={{ item.tty }}{% endif %}
-        {% if item.network is defined %}--network={{ item.network }}{% endif %}
-        {% if item.ip is defined %}--ip={{ item.ip }}{% endif %}
-        {% if item.etc_hosts is defined %}
-        {% for i, k in item.etc_hosts.items() %}{% if i != item.name %}--add-host {{ i }}:{{ k }} {% endif %}{% endfor %}{% endif %}
-        {% if item.hostname is defined %}--hostname={{ item.hostname }}{% elif item.name is defined %}--hostname={{ item.name }}{% endif %}
-        {% if item.systemd is defined %}--systemd={{ item.systemd | string | lower }}{% endif %}
-        {{ item.extra_opts | default([]) | join(' ') }}
-        {{ item.pre_build_image | default(false) | ternary('', 'molecule_local/') }}{{ item.image }}
-        {{ (command_directives_dict | default({}))[item.name] | default('') }}
+      containers.podman.podman_container:
+        name: "{{ item.name }}"
+        cap_add: "{{ item.capabilities | default(omit) }}"
+        # it's necessary to exclude empty args as ansible modules run_command method doesn't handle this well
+        cmd_args: "{{ molecule_podman_args | reject('equalto', '') | list + item.extra_opts | default([]) }}"
+        command: "{{ (command_directives_dict | default({}))[item.name] | default('') }}"
+        detach: "{{ item.detach | default(omit) }}"
+        device: "{{ item.devices | default(omit) }}"
+        dns: "{{ item.dns_servers | default(omit) }}"
+        env: "{{ item.env | default(omit) }}"
+        etc_hosts: "{{ item.etc_hosts | default(omit) }}"
+        executable: "{{ podman_exec }}"
+        expose: "{{ item.exposed_ports | default(omit) }}"
+        hostname: "{{ item.hostname | default(omit) }}"
+        image: "{{ item.pre_build_image | default(false) | ternary('', 'molecule_local/') }}{{ item.image }}"
+        ip: "{{ item.ip | default(omit) }}"
+        network: "{{ item.network | default(omit) }}"
+        pid: "{{ item.pid_mode | default(omit) }}"
+        privileged: "{{ item.privileged | default(omit) }}"
+        publish: "{{ item.published_ports | default(omit) }}"
+        security_opt: "{{ item.security_opts | default(omit) }}"
+        systemd: "{{ item.systemd | default(omit) }}"
+        tmpfs: "{{ item.tmpfs | default(omit) }}"
+        tty: "{{ item.tty | default(omit) }}"
+        ulimits: "{{ item.ulimits | default(omit) }}"
+        volume: "{{ item.volumes | default(omit) }}"
+        state: started
       register: server
+      changed_when: false
```

Comment:
@juanvalino 

I think `changed_when: false` should be removed.

The change status of `Create molecule instance(s)` will show in `Wait for instance(s) creation to complete`.

If it is always `false`, first time molecule creates the container, the change will show the wrong status.

```suggestion
```

Reply:
Thanks for your explanation.

> When containers are created, task Create molecule instance(s) return unchaged, but next task Wait for instance(s) creation to complete return changed. As final user you see something was created

I am afraid it does not work as you expected. Can you test it?

If `changed_when: false` is added, the change will always be false, even it is the first time to create the container.


Here is my test playbook for your reference:

No matter `/tmp/a` and `/tmp/b` exist or not, the change will always be false if `changed_when: false` is set.

```shell
touch /tmp/a /tmp/b
ansible-playbook test.yaml
```

```yaml
---
- name: Check if variable is defined and not empty
  hosts: localhost
  gather_facts: false

  tasks:
    - name: Asynchronous yum task
      ansible.builtin.file:
        path: "/tmp/{{ item }}"
        state: absent
      async: 1000
      poll: 0
      register: rst
      changed_when: false
      loop:
        - a
        - b

    - name: debug
      ansible.builtin.debug:
        msg: "{{ item }}"
      loop:
        "{{ rst.results }}"

    - name: Wait for asynchronous job to end
      ansible.builtin.async_status:
        jid: '{{ item.ansible_job_id }}'
      register: job_result
      until: job_result.finished
      retries: 100
      delay: 10
      changed_when: item is changed
      loop:
        "{{ rst.results }}"
```